### ISSUE TITLE
Units Disregard Owned Mines 2.0

### ISSUE THREAD

**Issue by RadicalEdward2** [OpenRA/OpenRA#13291]
I made a post back in February about [units disregarding mines](https://github.com/OpenRA/OpenRA/issues/12835). The post was closed because it was going to be fixed in (at the time) the next update. 

Player-owned units still run over player-owned mines (belonging to same person that laid the mines).

**Comment by MustaphaTR** [comment]
This is because it was reverted in #12967.

**Comment by RadicalEdward2** [comment]
I don't understand...
So they... want there to be the inconvenience of clumsy pathfinding resulting in friendly mine deaths? 
I'm not getting what back peddling was for.

**Comment by CH4Code** [comment]
Yes - it is a feature -  better not bury any mines in your own backyard.

**Comment by cjshmyr** [comment]
This is probably one of the worst gotchas the ra mod has. RE: [the discussion of why it got reverted](https://github.com/OpenRA/OpenRA/pull/12337), I assume firing on your own mines (or can we even just force-move?) will remove any accidental walled in problems.

**Comment by RadicalEdward2** [comment]
I still don't get the reason behind allowing your own units to choose death over common sense. I mean, the AI doesn't spawn mine layers anyway so there isn't really anyone benefiting from this "balance". I never use minelayers because I know my units pathfinding will somehow get them to run them over.

It's like having a gun in a household with children and choosing not to take safety precautions (like locking it in a safe or teaching  them gun safety) when you know for a fact the brats will try to play with it. 
Except in this case, they can actually prevent an incident from happening but, opted not to because it _wouldn't be fair that they didn't get to learn from the mishap_ (even if it kills them before they could apply the experience elsewhere). 

I'm sorry but, if someone has enough time to make a game of lemmings out of blocking paths with mines, there's just something wrong with that individual (considering the deploy key doesn't work so you'd have to manually click the minelayer ever...single...time).

**Comment by CH4Code** [comment]
FYI, you can place an extended minefield by using "crtl"-button.

 Its just risky to place mines in paths where your units will obviously move, and requires attention of the player to avoid move commands trough that region. (Good thing to require attention to your actions in an RTS) 
It would be ridiculous if you could put mines everywhere inside your base, your units avoiding the automatically. No commander ever in history put fucking MINES in his OWN base...

If you really, really want mines defending your base try this: 
![minefield-defense](https://cloud.githubusercontent.com/assets/16046997/26275973/8237b318-3d6d-11e7-8b16-8debabee1817.png)

Only enemy vehicles can crush your fences, so none of your units goes into that fenced area.

**Comment by GraionDilach** [comment]
>It would be ridiculous if you could put mines everywhere inside your base, your units avoiding the automatically. No commander ever in history put fucking MINES in his OWN base...

Sounds like you never played Generals. China can deploy minefields around buildings as a method of defense/infiltration/stealth detection. Zero Hour even added the neutron mine upgrade which made trespassing enemy vehicles emptied out.

**Comment by RadicalEdward2** [comment]
> Sounds like you never played Generals. China can deploy minefields around buildings as a method of defense/infiltration/stealth detection. Zero Hour even added the neutron mine upgrade which made trespassing enemy vehicles emptied out.

That's true. I didn't ever play Generals. Mostly because I want more Tiberium or Red Alert story; not a third story for me to get lost in. Also, my computer (at the time) wasn't strong enough (Gateway Windows 98) lol

Being able to press ctrl to place lines of mines is news to me though. I thought the highlighted tile was just artifacting left behind that wasn't fixed yet. I'm used to holding multiple keys. ctrl-D for defend, D for deploy, etc.
